api/src/main/java/org/openmrs/module/referencedemodata/patient/DemoPatientGenerator.java

```diff
@@ -200,13 +228,11 @@ private Patient createDemoPatient(PatientService ps, PatientIdentifierType patie
 		Patient patient = createBasicDemoPatient(patientIdentifierType, location, visitCount);
 		patient = ps.savePatient(patient);
 		
-		VisitService vs = Context.getVisitService();
 		Visit lastVisit = null;
 		for (int i = 0; i < visitCount; i++) {
 			boolean shortVisit = randomDoubleBetween(0, 1) < 0.8d;
-			Visit visit = createDemoVisit(patient, vs.getAllVisitTypes(), location, shortVisit, lastVisit,
+			Visit visit = createDemoVisit(patient, getVisitService().getAllVisitTypes(), location, shortVisit, lastVisit,
 					visitCount - (i + 1));
-			vs.saveVisit(visit);
```

Comment:
I'm not sure I understand why this was moved from this function? If we need the visit saved before doing something than surely that's just a separate step that should come after we create the visit and not part of creating the demo visit right?

Reply:
Saving a visit in  the `createDemoVisit()` method seems a more precise way of doing it. This is also because appointment and program generation is part of the visit creation stuff, following your suggestion in the previous review,
>get to doing this eventually with the visit and encounter generation stuff